djangosaml2/views.py

```diff
@@ -250,14 +250,18 @@ def get(self, request, *args, **kwargs):
 
         # SSO options
         sign_requests = getattr(conf, '_sp_authn_requests_signed', False)
-        if sign_requests:
-            sso_kwargs["sigalg"] = getattr(conf, '_sp_signing_algorithm',
-                                           saml2.xmldsig.SIG_RSA_SHA256
-            )
-            sso_kwargs["digest_alg"] = getattr(conf,
-                                               '_sp_digest_algorithm',
-                                               saml2.xmldsig.DIGEST_SHA256
-            )
+        csc = getattr(settings, 'SAML_CONFIG', {}).get(
+                'service',
+                {}
+            ).get('sp', None)
+        if csc:
+            if sign_requests:
+                sso_kwargs["sigalg"] = csc.get('signing_algorithm',
+                                            saml2.xmldsig.SIG_RSA_SHA256)
+                sso_kwargs["digest_alg"] = csc.get('digest_algorithm',
+                                                saml2.xmldsig.DIGEST_SHA256)
+            sso_kwargs['requested_authn_context'] = csc.get('requested_authn_context', None)
```

Comment:
Here you MUST build the AuthnContext object, until pysaml2 won't merge your PR

Reply:
I was expecting user to add the `RequestedAuthnContext` as following in SAML_CONFIG

```python
...
'requested_authn_context': saml2.samlp.RequestedAuthnContext(
    authn_context_class_ref=[
        saml2.saml.AuthnContextClassRef(
            saml2.saml.AUTHN_PASSWORD_PROTECTED
        ),
    ],
    comparison="exact",
),
...
```

How do you expect user to set this configuration?

Reply:
'requested_authn_context': {
    'authn_context_class_ref': saml2.saml.AUTHN_PASSWORD_PROTECTED, 
    'comparison': "exact"
}

@c00kiemon5ter what do you think about this solution also for PySAML2 configuration?

Reply:
@peppelinux should I make these changes or wait for Ivan's review?

Reply:
Go ahead, Life Is youth!
We'll handle this in the way we decided, @c00kiemon5ter can Always make a good clue and we'll see if change again or not but, for now, let's get the things done!